packages/react-storage/src/components/StorageBrowser/actions/useAction/createEnhancedListHandler.ts

```diff
@@ -15,9 +15,10 @@ interface SearchOptions<T> {
   query: string;
   /**
    * The key of the object in the list to filter by, which must have a string value.
+   * Also accepts a function which takes an item and returns the appropriate key.
    * This determines where the `query` will be searched.
    */
-  filterBy: KeyWithStringValue<T>;
+  filterBy: KeyWithStringValue<T> | ((item: T) => KeyWithStringValue<T>);
```

Comment:
Locations View falls back to using bucket for the "prefix" column in the view if prefix isn't available. This lets users provide a fallback.